datastream-server-api/src/main/java/com/linkedin/datastream/server/FlushlessEventProducerHandler.java

```diff
@@ -70,7 +70,7 @@ public void send(DatastreamProducerRecord record, String source, int sourceParti
     status.register(sourceCheckpoint);
     _eventProducer.send(record, ((metadata, exception) -> {
       if (exception != null) {
-        LOG.error("Failed to send datastream record: " + metadata, exception);
+        LOG.debug("Failed to send datastream record: " + metadata, exception);
```

Comment:
Is it possible to expose whether we're on shutdown path and only then print this as debug?
If not, please ensure that there are logs elsewhere to catch send failures from logs.

Reply:
The caller of this method has the shutdown path check and prints it otherwise.